VideoLocker/res/layout/fragment_discussion_thread_posts.xml

```diff
@@ -1,66 +1,79 @@
+                android:textColor="@color/edx_grayscale_neutral_base" />
 
-        <android.support.v7.widget.AppCompatSpinner
-            style="?attr/popupSpinnerStyle"
-            android:id="@+id/discussion_posts_sort_spinner"
-            android:layout_width="wrap_content"
-            android:layout_height="match_parent"
-            android:minWidth="@dimen/discussion_posts_filter_popup_width"
-            tools:listitem="@layout/row_discussion_thread_dropdown"
-            tools:targetApi="17" />
+            <android.support.v7.widget.AppCompatSpinner
+                android:id="@+id/discussion_posts_filter_spinner"
+                style="?attr/popupSpinnerStyle"
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:layout_marginLeft="@dimen/edx_margin"
+                android:layout_marginStart="@dimen/edx_margin"
+                android:minWidth="@dimen/discussion_posts_filter_popup_width"
+                tools:listitem="@layout/row_discussion_thread_dropdown"
+                tools:targetApi="17" />
 
-    </LinearLayout>
+            <android.support.v7.widget.AppCompatSpinner
+                android:id="@+id/discussion_posts_sort_spinner"
+                style="?attr/popupSpinnerStyle"
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:minWidth="@dimen/discussion_posts_filter_popup_width"
+                tools:listitem="@layout/row_discussion_thread_dropdown"
+                tools:targetApi="17" />
 
-    <View style="@style/gray_separator" />
+        </LinearLayout>
 
-    <ListView
-        android:id="@+id/discussion_posts_listview"
-        android:layout_width="match_parent"
-        android:layout_height="0dp"
-        android:layout_weight="1"
-        tools:listitem="@layout/row_discussion_thread"/>
+        <Space style="@style/gray_separator" />
+
+        <ListView
+            android:id="@+id/discussion_posts_listview"
+            android:layout_width="match_parent"
+            android:layout_height="0dp"
+            android:layout_weight="1"
+            android:visibility="invisible"
+            tools:listitem="@layout/row_discussion_thread"
+            tools:visibility="visible" />
+
+        <include layout="@layout/create_new_item_layout" />
+
+    </LinearLayout>
 
     <TextView
         android:id="@+id/center_message_box"
         style="@style/content_unavailable_error_text"
-        android:layout_width="match_parent"
-        android:layout_height="0dp"
-        android:layout_weight="1"
-        tools:text="Error text comes here." />
+        tools:text="Error text comes here."
+        tools:visibility="visible" />
 
-    <include layout="@layout/create_new_item_layout" />
+    <include
+        layout="@layout/loading_indicator"
+        tools:visibility="visible" />
 
-</LinearLayout>
+</FrameLayout>
```

Comment:
@miankhalid @1zaman Dunno who made most of the discussion XML files but they are super easy to go through in the Android Studio preview since all the items are filled with the tools :)